### DISCUSSION TITLE
Windows compatibility issues?

### DISCUSSION THREAD

**Discussion by FelixZett** [Alexandre-Fernandez/astro-i18n#18] [Q&A]
It seems like I'm running into windows compatibility problems. What I did was the following:

- `npm create astro@latest`
- `npm install astro-i18n`
- `node_modules/.bin/astro-i18n install`
- `npm run i18n:sync`

Which gives me the following error:

`SyntaxError: Invalid regular expression: /D:\projects\steadfast-satellite\src\pages\(?=(en$)|(en\(.+)?))/: Unmatched ')'`

The problem should be easily reproducible with the mentioned commands.
Am I doing something wrong?

**Comment by Alexandre-Fernandez** [comment]
Thanks for reporting, I'll try fixing this today, unfortunately I don't have a windows machine to test it but I know why this is happening.

**Reply by FelixZett** [reply]
I can easily test anything if you provide it on a branch or something 👍

**Reply by Alexandre-Fernandez** [reply]
I think the bug happens in this function :
https://github.com/Alexandre-Fernandez/astro-i18n/blob/05452a8524574b356aa2cf509e58d3103cb35258/src/core/fs/index.ts#L295
because `${sep}`, in windows is a backslash, but in regexes the backslash escapes the parenthesis, it should work with double backslash.
```ts
function regexPathStartsWith(string: string, base = "") {
	const regexSep = sep === "\\" ? "\\\\" : sep
	let prefix = base
	if (prefix.at(-1) !== sep) prefix = `${prefix}${regexSep}`
	return new RegExp(`${prefix}(?=(${string}$)|(${string}${regexSep}(.+)?))`)
}
```
Something like this should work... If you can test it that would be nice.

EDIT: you'll need to modify this for the `regexPathNotStartsWith` function aswell, it's just above.

**Reply by FelixZett** [reply]
It might take me a moment longer to test this. I am not familiar with modifying npm packages locally. I will try my best and get back to you, probably not today though.

**Reply by Alexandre-Fernandez** [reply]
I'll patch this and push it to main today, if it's not working on windows right now, worst case it still won't work.

**Reply by FelixZett** [reply]
That's even better! I will test it then and report back to you

**Reply by Alexandre-Fernandez** [reply]
please try with `1.6.4` and tell me if you encounter any problems

**Reply by FelixZett** [reply]
It works! Thank you very much for fixing this so quickly
